src/transformers/pipelines/text2text_generation.py

```diff
@@ -345,11 +363,13 @@ class TranslationPipeline(Text2TextGenerationPipeline):
     # Used in the return key of the pipeline.
     return_name = "translation"
 
-    def check_inputs(self, input_length: int, min_length: int, max_length: int):
-        if input_length > 0.9 * max_length:
+    def check_inputs(self, input_length: int, min_length: int, max_new_tokens: int):
+        # Updated parameter name from max_length to max_new_tokens to align with modern generation practices
+        # This method validates that input length doesn't exceed 90% of the maximum new tokens to generate
+        if input_length > 0.9 * max_new_tokens:
             logger.warning(
-                f"Your input_length: {input_length} is bigger than 0.9 * max_length: {max_length}. You might consider "
-                "increasing your max_length manually, e.g. translator('...', max_length=400)"
+                f"Your input_length: {input_length} is bigger than 0.9 * max_new_tokens: {max_new_tokens}. You might consider "
+                "increasing your max_new_tokens manually, e.g. translator('...', max_new_tokens=400)"
```

Comment:
If we're using `max_new_tokens` this check can be removed entirely! It was mostly relevant when `max_length` =~ `input_length`, which usually meant the user had set the values badly.